packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -481,20 +482,16 @@ export class ConnectionManager implements IConnectionManager {
                     connection = undefined;
                 }
             } catch (origError) {
-                if (typeof origError === "object" && origError !== null &&
-                    origError?.errorType === DeltaStreamConnectionForbiddenError.errorType) {
+                const error = normalizeError(origError);
+                copyRetryProps(error, origError);
```

Comment:
I think copyRetryProps() is not strictly required today (as all errors we throw in driver should be "our" errors, and thus normilizeError() should not do anything with it). Possibly validating same object is returned all is required here (and possibly recording and not using it if it's not "ours").

On other hand, if we assume that we can't rely on that (i.e., errors are not ours), then maybe we should not trust other properties on it? We made a call not to log them in such way, but it's not very clear why we would trust it in one place and not trust in another. PII violations are more visible for sure, but this code crashing (if format of those properties is not what we expect) or hanging for long time (if data fits expected format but has different meaning) is not great.

Clearly existing code is vulnerable to the same degree, so it's more of a realization / talking aloud about potential problem.
What do you think?

Reply:
Yeah, interesting. I think what you're getting at is that `canRetryOnError` is not as paranoid as our other code (telemetry) that deals with unknown errors.  We could check if it's "our" error before pulling off `canRetry`, but I think in practice I'm not worried about it.

Partly not worried because in the future where we don't throw our own errors, then we'll see the breadth of different types of errors that can fail here, and some (if not all) will have a `canRetry` property on the type, and it won't be an issue.

Reply:
Related: https://github.com/microsoft/FluidFramework/pull/8971/files

In this case, we're counting on pulling `canRetry` off an error from some other code.  This also should be updated such that the token provider returns a strongly-typed failure object if it doesn't succeed.